### PR TITLE
feat(bank-holidays): add week-end and bank holidays

### PR DESCRIPTION
**Pull request by Steffy29**
add week-end and bank holidays on calendar displayed for portability
to disable the possibility to select them and generate failures

- [x] Linked with PR https://github.com/ovh-ux/telecom-universe-components/pull/45

Ref: UXCT-19

### CODE REVIEW

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js [hunk 1]
```diff
@@ -1,242 +1,255 @@
...
-    }).catch((err) => {
-      TucToast.error([$translate.instant('telephony_alias_portability_order_error'), _.get(err, 'data.message')].join(' '));
-      return $q.reject(err);
-    }).finally(() => {
-      self.order.isOrdering = false;
+      self.onSDATypeChange = function () {
+        self.order.socialReason = self.isSDA ? 'corporation' : 'individual';
+      };
+
+      self.openDesireDatePicker = function (ev) {
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: I know that `ev` stands for event, but it should be not abbreviated

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js [hunk 2]
```diff
@@ -1,242 +1,255 @@
...
+        if (number) {
+          number = number.replace(/^\+/, '00');
+        }
+        return number;
+      };
+
+      self.goToConfigStep = function () {
+        self.order.addressTooLong = false;
+
+        if ((_.get(self.order, 'streetName', '').length + _.get(self.order, 'streetNumber', '').length + _.get(self.order, 'streetNumberExtra', '').length + _.get(self.order, 'streetType', '').length) >= 35) {
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: You can do this this way if you want : 
```suggestion
        if (`${_.get(self.order, 'streetName', '')}${_.get(self.order, 'streetNumber', '')}${_.get(self.order, 'streetNumberExtra', '')}${_.get(self.order, 'streetType', '')}`.length >= 35) {
```

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js [hunk 3]
```diff
@@ -1,242 +1,255 @@
...
+        params.desireDate = moment(params.desireDate).format('Y-MM-DD');
+
+        return params;
+      };
+
+      self.fetchPriceAndContracts = function () {
+        self.step = 'summary';
+        return OvhApiOrder.Telephony().v6()
+          .getPortability(self.getOrderParams()).$promise
+          .then((result) => {
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: You can destructure `result` :  
```suggestion
          .then(({details, contracts, prices}) => {
            self.details = details;
            self.contracts = contracts;
            self.prices = prices;
```

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js [hunk 4]
```diff
@@ -1,242 +1,255 @@
...
+            TucToast.error(_.get(err, 'data.message'));
+            return $q.reject(err);
+          });
+      };
+
+      self.submitOrder = function () {
+        self.order.isOrdering = true;
+        return OvhApiOrder.Telephony().v6().orderPortability({
+          billingAccount: self.order.billingAccount,
+        }, _.omit(self.getOrderParams(), 'billingAccount')).$promise.then((result) => {
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: Same thing, you can destructure `result`
```suggestion
        }, _.omit(self.getOrderParams(), 'billingAccount')).$promise.then(({orderId, url}) => {
```

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js [hunk 5]
```diff
@@ -1,242 +1,255 @@
...
+            () => {
+              // in this case it means that nic bill and connected are not the same
+              // so display a message telling that order must be validated by clicking
+              // no need to reject because portablity order is a success
+              // and validation can always be done by clicking
+              self.order.autoPay = false;
+            },
+          );
+        }).catch((err) => {
+          TucToast.error([$translate.instant('telephony_alias_portability_order_error'), _.get(err, 'data.message')].join(' '));
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]:
```suggestion
          TucToast.error(`${$translate.instant('telephony_alias_portability_order_error')} ${_.get(err, 'data.message')});
```

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js [hunk 6]
```diff
@@ -1,242 +1,255 @@
...
-          label: $translate.instant(`telephony_order_specific_typology_${typo.replace(new RegExp('^be_|fr_'), '')}_label`),
+angular.module('managerApp')
+  .controller('TelecomTelephonyAliasPortabilityOrderCtrl',
+    function ($scope, $stateParams, $translate, $q, moment, TelephonyMediator,
+      OvhApiMe, OvhApiOrder, TucBankHolidays, TucToast) {
+      const self = this;
+
+      const specialNumberPrefix = {
+        france: ['00338'],
+        belgium: ['0032800', '003278', '0032900', '003270'],
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: It was previous code

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js [hunk 7]
```diff
@@ -1,242 +1,255 @@
...
+        }, true);
+
+        $scope.$watch('PortabilityOrderCtrl.isSDA', () => {
+          self.order.offer = self.isSDA ? 'company' : 'individual';
+        }, true);
+
+        self.isSpecialNumber = false;
+        self.typologies = { france: [], belgium: [] };
+
+
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: same here

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js (hunk 1 above)
Comment (Steffy29) [reply client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: previous code

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js [hunk 8]
```diff
@@ -1,242 +1,255 @@
...
+                || (isBankHoliday));
+          },
+        };
+
+        // fetch list of billing accounts
+        return TelephonyMediator.getAll().then((groups) => {
+          self.billingAccounts = groups;
+          self.order.billingAccount = $stateParams.billingAccount;
+        }).catch((err) => {
+          TucToast.error(_.get(err, 'data.message'));
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: previous code

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js (hunk 2 above)
Comment (Steffy29) [reply client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: previous code

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js (hunk 3 above)
Comment (Steffy29) [reply client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: previous code

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js (hunk 4 above)
Comment (Steffy29) [reply client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: previous code

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js (hunk 5 above)
Comment (Steffy29) [reply client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: previous code

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (JeremyDec) [reply client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: Even if it is a refactor, and like you said, it is not your fresh new code, we should improve it :).
Keep in mind the "boy-scout rule" : "Leave your code better than you found it." (see https://deviq.com/boy-scout-rule/).

File: client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js [hunk 9]
```diff
@@ -1,242 +1,249 @@
...
+          self.order.isContractsAccepted = false;
+        });
+
+        $scope.$watch('PortabilityOrderCtrl.order', () => {
+          self.order.contactName = self.order.name;
+        }, true);
+
+        $scope.$watch('PortabilityOrderCtrl.isSDA', () => {
+          self.order.offer = self.isSDA ? 'company' : 'individual';
+        }, true);
```
Comment (marie-j) [inline client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: I'm quite curious on why those are necessary and using a `ng-model` and listening to events such as `ng-change` like it is already partially done won't do 
https://github.com/ovh-ux/ovh-manager-telecom/blob/881f41fc439e5b2bea9f8280c8d7e38524a361d5/client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.html#L85

Comment (Steffy29) [reply client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js]: I don't know why it was coded like that.

**Comment by antleblanc** [comment]
@Steffy29 you are now able to upgrade to the latest telecom-universe-components.

ref: https://github.com/ovh-ux/telecom-universe-components/blob/master/CHANGELOG.md#140-2018-12-31

File: package.json [hunk 10]
```diff
@@ -42,7 +42,7 @@
     "@bower_components/crypto": "anodynos/node2web_crypto#3.46.1",
     "@bower_components/international-phone-number": "mareczek/international-phone-number#^0.0.16",
     "@bower_components/matchmedia": "paulirish/matchMedia.js#~0.2.0",
-    "@ovh-ux/telecom-universe-components": "^1.3.0",
+    "@ovh-ux/telecom-universe-components": "git+https://github.com/ovh-ux/telecom-universe-components.git#feat/holidays",
```
Comment (antleblanc) [inline package.json]: @Steffy29 you are now able to upgrade to the latest telecom-universe-components.

ref: https://github.com/ovh-ux/telecom-universe-components/blob/master/CHANGELOG.md#140-2018-12-31

**Comment by ovh-cds** [comment]
CDS Report build-manager#1140.0 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✘

**Pull request merged**